### PR TITLE
feat(pymilvus/settings.py): Load configuration without altering the environment

### PR DESCRIPTION
**Pull request by laipz8200**
related #2166.

This PR will replace `environs` with the more lightweight `python-dotenv`. It can load configuration from the dot env file without altering the system environment.

### CODE REVIEW

**Review by XuanYang-cn — CHANGES_REQUESTED** [review]
Please replace the dependencies in pyproject.toml too.

**Review by XuanYang-cn — APPROVED** [review]

**Comment by laipz8200** [comment]
Hello @XuanYang-cn, will this be merged? What should I do?

File: pymilvus/settings.py [hunk 1]
```diff
@@ -1,21 +1,17 @@
...
 import logging.config
 
-import environs
+from dotenv import dotenv_values
 
-env = environs.Env()
-
-with contextlib.suppress(Exception):
-    env.read_env(".env")
+env_dict = dotenv_values(".env")
```
[the remark this answers is not in the document]
Comment (laipz8200) [reply pymilvus/settings.py]: Thank you for your suggestion. I previously misunderstood the behavior of `load_dotenv`, but I will use it now.

File: pyproject.toml [hunk 2]
```diff
@@ -21,7 +21,7 @@ dependencies=[
     "setuptools<70.1;python_version<='3.8'",
     "grpcio>=1.49.1, <=1.63.0",
     "protobuf>=3.20.0",
-    "environs<=9.5.0",
+    "python-dotenv==1.0.1",
```
[the remark this answers is not in the document]
Comment (laipz8200) [reply pyproject.toml]: Sorry about that. I changed it to `>=1.0.1, <2.0.0`

**Comment by XuanYang-cn** [comment]
Sorry about the late response!
/lgtm
/approve

**Comment by laipz8200** [comment]
@XuanYang-cn Thank you:)

**Comment by sloria** [comment]
fwiw, `python-dotenv`'s `load_env` _does_ alter `os.environ`. in fact, `environs` [delegates to `load_env`](https://github.com/sloria/environs/blob/88d9df372ad933d2b90e2e29461911b74fae8d1b/src/environs/__init__.py#L414-L421), so this PR doesn't really change that.

i'm planning to add support for reading `.env` without altering `os.environ` https://github.com/sloria/environs/issues/393 .

btw this isn't to argue againt this PR getting merged--i'm all for removing dependencies when sensible, even if they're my libraries 😅 . just raising that this PR may not have achieved its original intent

**Pull request merged**